nbgitpuller/pull.py

```diff
@@ -303,14 +303,14 @@ def main():
 
     parser = argparse.ArgumentParser(description='Synchronizes a github repository with a local repository.')
     parser.add_argument('git_url', help='Url of the repo to sync')
-    parser.add_argument('branch_name', default=None, help='Branch of repo to sync', nargs='?')
-    parser.add_argument('repo_dir', default='.', help='Path to clone repo under', nargs='?')
+    parser.add_argument('--branch_name', default=None, required=False, help='Branch of repo to sync', nargs='?')
```

Comment:
```suggestion
    parser.add_argument('--branch-name', default=None, required=False, help='Branch of repo to sync')
```

Reply:
I think flags starting with `--` by default are not required and set default to None, so maybe those are not required?

Reply:
```suggestion
    parser.add_argument('branch_name', default=None, required=False, help='Branch of repo to sync', nargs='?')
```

`branch_name` will make translate to `--branch-name` flag, and make the `branch_name` variable accessible from parsed args.

Reply:
+1 for consitently using python_snake_case and letting the --python-snake-case flag be assumed as a result.

Reply:
ah, ok! I wasn't aware of this behavior! I've resolved this suggestion

Reply:
### Conclusion: i got things wrong, ignore me

Without `--` added, it will be interpreted as a positional argument it seems, so adding `--` is relevant, but if its added, then we should use `-` as a separator instead of `_` like @yuvipanda suggested as its whats common across the JupyterHub org.

### Things I started writing when confused

I've gotten a bit confused now, so I'll write a summary to capture the situation.

---

__This is how it was:__

```
parser.add_argument('branch_name', ...
```

__This is the PR modification, to my knowledge unrelated to the PRs purpose:__

```
parser.add_argument('--branch_name', ...
```

__This is @yuvipanda suggestion:__

```
parser.add_argument('--branch-name', ...
```

And my suggestion is to let it be as it was, which I think will make a flag accessible just like in yuvi's suggestion (`--branch-name`). This should be documented... https://docs.python.org/3/library/argparse.html

Reply:
OK got it so "-" as separator but we are worried about the breaking change so we will only do this with repo-dir but call it target-dir. Good deal